src/dataflow/components/dataflow-program-topbar.tsx

```diff
@@ -12,43 +12,67 @@ interface TopbarProps {
   onProgramTimeSelectClick: (type: number) => void;
   isRunEnabled: boolean;
   runningProgram: boolean;
+  remainingTime: number;
   readOnly: boolean;
 }
 
 export const DataflowProgramTopbar = (props: TopbarProps) => {
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
     props.onProgramTimeSelectClick(Number(event.target.value));
   };
+  const runTime = props.programRunTimes.find( (rt: ProgramRunTime) => rt.val === props.programDefaultRunTime );
+  const remainingHours = Math.floor(props.remainingTime / 3600);
+  const remainingMinutes = Math.floor((props.remainingTime - remainingHours * 3600) / 60);
+  const remainingSeconds = (props.remainingTime - (remainingHours * 3600) - (remainingMinutes * 60) ) % 60;
+  const progressWidth = (76 * props.remainingTime / props.programDefaultRunTime).toString() + "px";
+  const hoursString = String(remainingHours).padStart(2, "0");
+  const minutesString = String(remainingMinutes).padStart(2, "0");
+  const secondsString = String(remainingSeconds).padStart(2, "0");
```

Comment:
Are we concerned about `padStart()` availability in browsers? We can probably live without IE at this point. In the absence of specific minimum system requirements, maybe Chrome 57/Edge 15/Firefox 48/Safari 10 is sufficient.

Reply:
Yeah, we're probably going to be safe, but something to keep an eye on.